src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminalAccessibleBuffer.ts

```diff
@@ -130,6 +132,13 @@ export class AccessibleBufferWidget extends DisposableStore {
 					break;
 			}
 		}));
+		this.add(this._editorWidget.onDidChangeCursorPosition((c) => {
+			if (c.source === 'mouse' || c.source === 'keyboard') {
+				this._cursorPosition = { lineNumber: c.position.lineNumber, column: c.position.column };
+			} else if (this._cursorPosition) {
+				this._resetPosition();
+			}
+		}));
```

Comment:
@rebornix are we doing something wrong here by applying edits to the model as changes come in? We're needing to jump through a bunch of hoops to keep the cursor and scroll position when changes happen in the terminal buffer and we add them to the monaco instance.

Reply:
I think we might run into trouble in the future by listening to cursor position change and then resetting it. From what @Tyriar shared about how we ran into this problem, there are two alternatives:

* Before setting new value to the model, save the view state (which includes the cursor state). Once model is updated, restore the view state.
* Or we do a diff of the content, and apply edits incrementally, like what we did in text file editor https://insiders.vscode.dev/github/microsoft/vscode/blob/rebornix/integral-monkey/src/vs/editor/common/services/modelService.ts#L396-L397

Reply:
I have tried many approaches and believe this is the best solution. Let's see if there are issues before rethinking the strategy.